implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -270,14 +277,34 @@ async fn run_foreground_node(
 
     if let Some(path) = &cmd.launch_config {
         let node_opts = super::NodeOpts {
-            api_node: cmd.node_name,
+            api_node: cmd.node_name.clone(),
         };
         start_services(&ctx, &tcp, path, addr, node_opts, &opts).await?
     }
 
+    if cmd.watch_stdin {
+        stop_node_on_eof().await;
+    }
+
     Ok(())
 }
 
+// Read STDIN until EOF is encountered and then terminate the node
+async fn stop_node_on_eof() {
+    let reader = tokio::io::BufReader::new(tokio::io::stdin());
+    let mut lines = reader.lines();
+
+    loop {
+        match lines.next_line().await {
+            Ok(Some(_)) => (),
+            Ok(None) => break,
+            Err(_) => panic!("Something went wrong"),
+        }
+    }
+
+    std::process::exit(exitcode::IOERR);
```

Comment:
You can pass the `ockam::Context` as a variable to this function and then call `ctx.stop().await();`. That way the node will be gracefully stopped and exit the process without throwing an error.

Also, since this stops the process, you should do the equivalent to what's happening in the Stop command which is just calling the next function: `opts.state.nodes.get(&cmd.node_name)?.kill_process(false);`.